x/evm/keeper/state_transition.go

```diff
@@ -67,7 +67,15 @@ func (k Keeper) GetHashFn() vm.GetHashFunc {
 		case k.ctx.BlockHeight() == h:
 			// Case 1: The requested height matches the one from the context so we can retrieve the header
 			// hash directly from the context.
-			return common.BytesToHash(k.ctx.HeaderHash())
+			// Note: we cannot use k.ctx.HeaderHash() because the headerHash it set only at begin block
+			// so it won't be set in case of simulated context
+			contextBlockHeader := k.ctx.BlockHeader()
```

Comment:
we should check first `if len(k.ctx.HeaderHash()) == 0` to prevent hashing the header if it's not necessary

Reply:
sure, done